### ISSUE TITLE
an inconsistent error message for array size expressions

### ISSUE THREAD

**Issue by pirapira** [ethereum/solidity#1259]
At version `0.4.3-develop.2016.10.20+commit.9d304501.Linux.g++`, an array size can be a sum of constants, but cannot be a division by zero.  In case of division by zero, the error message says that the compiler expected an integer literal.

Either the index expression should be limited to an integer literal, or the error message should be changed to "expected a constant expression whose value is a non-negative integer."

```
contract T {
    function f() {
        int[3+4] memory a; //succeeds
    }
    function g() {
        int[3/0] memory a; // fails
    }
}
```

```
test.sol:6:13: Error: Invalid array length, expected integer literal.
        int[3/0] memory a;
            ^-^
```

**Comment by axic** [comment]
Should raise an error that division by zero is not a valid integer.

**Comment by wadeAlexC** [comment]
I'm working on this issue, and am a bit stuck. Looking for a push in the right direction. Here's the function where the error message is being called: 
https://github.com/ethereum/solidity/blob/010189d58eca560dd319aab07daa43bda0911a40/libsolidity/analysis/ReferencesResolver.cpp#L142

My initial thought was to move "if(!lengthType)" to a separate section, but that won't catch array size declarations like "fixed(3.5)". My next idea was to add a field to the class RationalNumberType, found here: 
https://github.com/ethereum/solidity/blob/develop/libsolidity/ast/Types.h#L396

Unfortunately, length is a null pointer, so that won't work either. The last thing I've found is the line where the actual division is taking place: 
https://github.com/ethereum/solidity/blob/develop/libsolidity/analysis/ConstantEvaluator.cpp#L48

I'm stuck at this point though - not sure where to go from here. I've already asked on gitter, and got nothing. Any pointers?
